lib/widgets/message_list.dart

```diff
@@ -511,60 +533,93 @@ class _UnreadMarker extends StatelessWidget {
   }
 }
 
-class StreamTopicRecipientHeader extends StatelessWidget {
-  const StreamTopicRecipientHeader({super.key, required this.message});
+class StreamMessageRecipientHeader extends StatelessWidget {
+  const StreamMessageRecipientHeader({super.key, required this.message, required this.showStream});
 
   final StreamMessage message;
+  final bool showStream;
 
   @override
   Widget build(BuildContext context) {
+    // For design specs, see:
+    //   https://www.figma.com/file/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=538%3A20849&mode=dev
+    //   https://github.com/zulip/zulip-mobile/issues/5511
     final store = PerAccountStoreWidget.of(context);
 
     final stream = store.streams[message.streamId];
-    final streamName = stream?.name ?? message.displayRecipient; // TODO(log) if missing
     final topic = message.subject;
-
-    final subscription = store.subscriptions[message.streamId];
-    final streamColor = Color(subscription?.color ?? 0x00c2c2c2);
+    final swatch = store.subscriptions[message.streamId]?.colorSwatch()
+      ?? _kFallbackStreamColorSwatch;
     final contrastingColor =
-      ThemeData.estimateBrightnessForColor(streamColor) == Brightness.dark
+      (ThemeData.estimateBrightnessForColor(swatch.barBackground) == Brightness.dark)
         ? Colors.white
         : Colors.black;
+    final textStyle = TextStyle(
+      color: contrastingColor,
+      fontFamily: 'Source Sans 3',
+      fontSize: 16,
+      letterSpacing: 0.02 * 16,
+      height: (18 / 16),
+    ).merge(weightVariableTextStyle(context, wght: 600, wghtIfPlatformRequestsBold: 900));
+
+    final streamWidget = (showStream)
+      ? GestureDetector(
+          onTap: () => Navigator.push(context,
+            MessageListPage.buildRoute(context: context,
+              narrow: StreamNarrow(message.streamId))),
+          child: Row(children: [
+            Padding(
+              // Figma shows 5px spacing but 6px here matches better visually.
+              padding: const EdgeInsets.symmetric(horizontal: 6, vertical: 11),
+              child: Icon(size: 18, color: swatch.iconOnBarBackground,
+                // A null [Icon.icon] makes a blank space.
+                (stream != null) ? iconDataForStream(stream) : null)),
+            Padding(
+              padding: const EdgeInsets.symmetric(vertical: 11),
```

Comment:
What's the origin of the number 11 for vertical padding? I'm not finding that in the Figma design.

Also, it seems like it'd be simpler to apply the 11px vertical padding to the entire Row. Is there a reason I'm missing why that wouldn't have the right effect?

Reply:
This is reverse engineered from a 40px height given the icon height (18p) for this element, and was also applied to the text widget that has 18px line height. The chevron has 16 height and thus given `vertical: 12`. I don't think it's clear that wrapping the whole row is an improvement as the chevron would need padding as well.

I'll take a closer look at the behavior of cranking up text size in the system settings. In your screenshot it seems that this doesn't affect icon sizes, in which case it might make sense to apply a uniform padding on text elements and vertically center the icons another way.

Reply:
After our call today I've settled on applying a `CrossAxisAlignment.center` to all the row elements and applying a `vertical: 11` padding to the text elements. I've changed the date line-height to match the text elements so they all visually appear to have the same baseline and let the icons settle on the center of the vertical (having also had to tweak the stream icon a bit).
The text elements maintain their alignment even if the system requests they be rendered larger, and the icons don't look terrible where they are placed:
![screen_enlarged_text](https://github.com/zulip/zulip-flutter/assets/98299/0334f025-77f4-436e-8bef-449dd71a1b6b)